src/v/cluster/members_frontend.cc

```diff
@@ -132,4 +138,79 @@ members_frontend::recommission_node(model::node_id id) {
     }
     co_return res.value().data.error;
 }
+
+ss::future<members_frontend::get_under_replicated_reply>
+members_frontend::get_under_replicated(model::node_id id) {
+    if (id != _self()) {
+        co_return get_under_replicated_reply{
+          .under_replicated = co_await is_node_under_replicated(id),
+          .error = errc::success};
+    }
+
+    constexpr model::node_id no_leader{-1};
+    absl::flat_hash_set<model::node_id> leaders;
+    const auto& parts = _partition_manager.local().partitions();
+    for (const auto& part : parts) {
+        leaders.insert(
+          _leaders.local().get_leader(part.first).value_or(no_leader));
+    }
+    if (leaders.count(no_leader)) {
```

Comment:
do you think that this condition is too strong? for example, if this node knows that a partition exists, but the partition doesn't have a leader for some reason or this node is unable to determine the leader...? im not sure what the answer is. we can always fine tune the semantics over time, too.